lib/active_model_serializers/key_transform.rb

```diff
@@ -0,0 +1,43 @@
+module ActiveModelSerializers
+  module KeyTransform
+    # Transforms keys to UpperCamelCase or PascalCase.
+    #
+    # @example:
+    #    "some_key" => "SomeKey",
+    # @see: active_support/inflector#camelize
+    def camel(hash)
+      hash.deep_transform_keys! { |key| key.to_s.camelize.to_sym }
+    end
+
+    # Transforms keys to camelCase.
+    #
+    # @example:
+    #    "some_key" => "someKey",
+    # @see: active_support/inflector#camelize
+    def camel_lower(hash)
+      hash.deep_transform_keys! { |key| key.to_s.camelize(:lower).to_sym }
+    end
+
+    # Transforms keys to dashed-case.
+    # This is the default case for the JsonApi adapter.
+    #
+    # @example:
+    #    "some_key" => "some-key",
+    # @see: active_support/inflector#dasherize
+    def dashed(hash)
+      hash.deep_transform_keys! { |key| key.to_s.dasherize.to_sym }
+    end
+
+    # Returns the hash unaltered
+    def unaltered(hash)
+      hash
+    end
+
+    module_function :camel, :camel_lower, :dashed, :unaltered
+
+    def self.method_missing(m, *args, &block)
+      warn "\"#{m}\" is not a defined key transform. Defaulting to :unaltered."
+      send(:unaltered, *args, &block)
```

Comment:
@remear @NullVoxPopuli So, before I asked:

> If I say `KeyTransform.micky_mouse` it should fail, right?

Here's a better example, I think:

`KeyTransform.caml`.  Person mistyped `camel` and now instead of a NoMethodError they're getting back `:unaltered`.  That's surprising.

I feel pretty strongly this should fail fast.  The warning message is good, but should be a failure message.  

Also, since you're defining the method missing on the module, are you also eating up all method missing calls on the `Adapter::Base`.  Just seems like asking for trouble with only benefit is a silent failure with a warning somewhere in the logs.